### PR TITLE
Weigh table panel feature

### PR DESCRIPTION
**Pull request by EugenKoulik**

### CODE REVIEW

File: Forms/DrawForm/DrawForm.cs [hunk 1]
```diff
@@ -86,6 +103,8 @@ private void MouseClickDrawForm(object sender, MouseEventArgs e)
                 {
                     vertexDraws.Add(vertexDraw);
 
+                    matrixWeightPanel.CreateMatrix(1);
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/DrawForm.cs]: Name does not match functionality

File: Forms/DrawForm/WeightTable.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+        private List<List<InputCountBox>> matrixx;
+
+        private WeightTable weightTable;
+     
+
+        public MatrixWeightPanel(WeightTable weightTable)
+        {
+
+            matrixx = new List<List<InputCountBox>>();
```
Comment (Maksim-Trolina) [inline Forms/DrawForm/WeightTable.cs]: *matrix

**Review by Maksim-Trolina — APPROVED** [review]

**Review by Maksim-Trolina — CHANGES_REQUESTED** [review]
Make

**Review by Maksim-Trolina — APPROVED** [review]

File: Forms/DrawForm/DrawForm.cs [hunk 3]
```diff
@@ -34,10 +35,17 @@ public partial class DrawForm : Form
...
 
+        private ToolPanel toolPanel;
+
+        private WeightTable weightTable;
+
+        private MatrixWeightPanel matrixWeightPanel;
+
         private Arrow arrow;
 
+        private List<List<InputCountBox>> matrix;
```
Comment (dnbss) [inline Forms/DrawForm/DrawForm.cs]: Why do you create the "private List<List<Input Count Box>> matrix" field if it is not used anywhere except in the constructor, where it is only initialized?

File: Forms/DrawForm/WeightTable.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+        public MatrixWeightPanel(WeightTable weightTable, List<List<InputCountBox>> matrix)
+        {
+
+            this.weightTable = weightTable;
+            this.matrix = matrix;
+
+        }
+
+        public void DrawingMatrix(int rows)
```
Comment (dnbss) [inline Forms/DrawForm/WeightTable.cs]: Why do you pass a parameter here, if you have a "matrix" field that already contains information about the number of elements? This is not safe.

**Review by dnbss — APPROVED** [review]

**Pull request merged**